src/shim.c

```diff
@@ -434,6 +441,13 @@ read_frame(struct cc_shim *shim)
 	} else {
 		free(buf);
 	}
+
+	shim_debug("Read frame HeaderLength: %d, Version: %d, Payloadlen: %d,"
```

Comment:
It would be easier to read this if you broke the format string so into chunks containing only 2 formats as that would align with the way you've paired up the variables below.